### DISCUSSION TITLE
Proper way to teleport an object

### DISCUSSION THREAD

**Discussion by SSNTails** [jrouwe/JoltPhysics#1041] [Q&A] [answered]
What's the correct way to teleport an object? I find that the below works for very minor teleports, but larger stuff messes up, making me think I need to refresh the broadphase, or do some other task as well.

`
	JPH::BodyInterface &bi = mPhysicsSystem->GetBodyInterface();
		bi.SetPositionAndRotation(object->collisionObjectID, mat.GetTranslation(), mat.GetRotation().GetQuaternion().Normalized(), JPH::EActivation::Activate);
`

**Comment by jrouwe** [comment, marked as answer]
This should be the correct way, if you look at the implementation of `SetPositionAndRotation` you'll see that it notifies the broadphase internally.

**Reply by SSNTails** [reply]
Thanks for verifying. Upon player death, I change its collision body to a sensor, and then back to a solid on respawn so that they fall through the world like when Mario dies in the old games. I wasn't removing the sensor property correctly on respawning back to the continue point. :P
